CONTRIBUTING.md

```diff
@@ -208,6 +208,45 @@ of the direction" suddenly has the appearance of "I approve of these changes."
 Instead, add a comment that mentions the usernames that you would like a review
 from. Ask explicitly what you would like feedback on.
 
+## Crate Creation
+
+If your PR includes a new crate, it is required to publish its v0.0.1 version
+before the PR can be merged.  Here're the steps:
+
+* Create a sub-directory for your new crate.
+* Under the newly-created directory, create a Cargo.toml file.  Below is an
+  example template:
+
+```
+[package]
+name = "solana-<PACKAGE_NAME>"
+version = "0.0.1"
+description = "<DESCRIPTION>"
+authors = ["Solana Maintainers <maintainers@solana.foundation>"]
+repository = "https://github.com/solana-labs/<REPO>"
+homepage = "https://solana.com/"
+documentation = "https://docs.rs/solana-<PACKAGE_NAME>"
+license = "Apache-2.0"
+edition = "2021"
+```
+
+* Submit the PR for initial review.  At this moment you should see the crate
+  check fails because the newly created crate has not yet known in public.
+
+* Once all review feedback has been addressed.  We will send you the token
+  for publishing crate for Solana.  Run the following command under your crate
```

Comment:
do we share the token to outside contributors 🤔 I think maybe we can publish the crate for them

Reply:
hmm, yeah maybe a better way is to first publish the crate with your own crates.io user and then transfer the ownership to the https://crates.io/users/solana-grimes user via a discussion on #releng

Reply:
Yep, we definitely shouldn't share the token with external contributors as it should be treated as our "private key".

I will include the above two options (publish the crate for them, or ownership transfer).

Reply:
I kinda think the external contributor path is better for "internal" contributors as well actually. Sharing that token around isn't exactly a best practice. 

Longer term we want to remove as much of the separation as possible between internal and  external contributor anyway

Reply:
Make sense.  Btw, when the crate is first published under a personal account, does the crate check verify whether it is under Solana or not?  Just want to make sure we don't accidentally merge some PR whose crate isn't published under Solana.

Reply:
yep, owner is checked: https://github.com/solana-labs/solana/blob/3d63f93bb0346886d0062576f3382bf0e0bc9572/ci/check-crates.sh#L24